File: quick-sort/quick-sort.js
const quicksort = arr => {
  if (arr.length < 2) return arr;

  const [pivot, ...rest] = arr;
  const less = rest.filter(x => x <= pivot);
  const greater = rest.filter(x => x > pivot);

  return [...quicksort(less), pivot, ...quicksort(greater)];
};

console.log(quicksort([15, 11, 10, 30, 20]));

File: quick-sort/quick_sort.py
def quick_sort(list):
    if len(list) < 2:
        return list
    else:
        pivot = list[0]
        less = [i for i in list[1:] if i <= pivot]
        greater = [i for i in list[1:] if i > pivot]
        return quick_sort(less) + [pivot] + quick_sort(greater)

print quick_sort([10])
print quick_sort([20, 10])
print quick_sort([15, 20, 10])